platform/macos/detect.py

```diff
@@ -239,8 +239,17 @@ def configure(env: "SConsEnvironment"):
 
     env.Append(LINKFLAGS=["-rpath", "@executable_path/../Frameworks", "-rpath", "@executable_path"])
 
+    if env["metal"] and env["arch"] != "arm64":
+        print("Metal is only supported on arm64.")
+        sys.exit(255)
+
+    if env["metal"]:
+        env.AppendUnique(CPPDEFINES=["METAL_ENABLED", "RD_ENABLED"])
+        env.Append(LINKFLAGS=["-framework", "Metal", "-framework", "MetalKit"])
+        env.Prepend(CPPPATH=["#thirdparty/spirv-cross"])
+
     if env["vulkan"]:
-        env.Append(CPPDEFINES=["VULKAN_ENABLED", "RD_ENABLED"])
+        env.AppendUnique(CPPDEFINES=["VULKAN_ENABLED", "RD_ENABLED"])
         env.Append(LINKFLAGS=["-framework", "Metal", "-framework", "IOSurface"])
```

Comment:
`metal` + `vulkan` will lead to passing `-framework Metal` twice. I assume this wasn't a problem in your testing?
Could possibly be improved with `AppendUnique` though with the way `-framework` and `Metal` are separate arguments, that wouldn't work so well I guess.

Reply:
It hasn't caused any issues with the build. Probably the best we could do is have an "extra_frameworks" `dict` and add to it for either Metal or Vulkan and then resolve it to a list of `-framework`, `<name>` pairs.

Reply:
@akien-mga I've fixed this by building a set and appending the unique list of frameworks after configuring `metal` and `vulkan`.